### PR TITLE
streamlit dashboard layout complete with email regexing

### PR DESCRIPTION
**Pull request by lukieh2014**
Streamlit dashboard layout set up with 4 pages, as described in #30 . User is able to input their email address and subscription preferences (alerts and/or daily summary reports) in the dashboard. Regex is used to verify and validate the user's email address.

Referring to ticket #48.

### CODE REVIEW

**Review by bentrzcinski — APPROVED** [review]
Looks great Luke, a really helpful template!

File: dashboard/dashboard.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+def is_valid_email(email: str) -> bool:
+    """Validates inputted user email address."""
+    email_regex = r'^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+    return re.match(email_regex, email) is not None
+
+
+def main_overview() -> None:
+    """Creates main overview page on dashboard."""
+    st.title("Main Overview Page")
```
Comment (ebradley12) [inline dashboard/dashboard.py:15]: Maybe the main title should include "Band Camp Tracker" somewhere and link to the website as well

File: dashboard/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    st.title("Report Download Page")
+    st.write("Download reports from this page.")
+
+
+def subscribe_page() -> None:
+    """
+    Creates subscribe page where users can subscribe
+    to daily summary PDF reports and/or alerts about
+    trends and genres via email.
```
Comment (ebradley12) [inline dashboard/dashboard.py:38]: Really nice docstrings

File: dashboard/dashboard.py (hunk 1 above)
Comment (lukieh2014) [reply dashboard/dashboard.py]: good idea, just added!

File: dashboard/dashboard.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+def is_valid_email(email: str) -> bool:
+    """Validates inputted user email address."""
+    email_regex = r'^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+    return re.match(email_regex, email) is not None
+
+
+def main_overview() -> None:
+    """Creates main overview page on dashboard."""
+    st.title("BandCamp Tracker: https://bandcamp.com/")
```
Comment (ebradley12) [inline dashboard/dashboard.py:15]: Nice change

**Pull request merged**